src/main/java/hudson/remoting/Channel.java

```diff
@@ -1734,7 +1736,7 @@ public static void dumpDiagnosticsForAll(@Nonnull PrintWriter w) {
          * @param channel the {@link Channel}.
          */
         private Ref(@CheckForNull Channel channel) {
-            this.name = channel.getName();
+            this.name = channel != null ? channel.getName() : "unknown (null reference)";
```

Comment:
FindBugs prints warnings, but it does not fail the build now. I am handling it as https://issues.jenkins-ci.org/browse/JENKINS-37566 . One year ago there were more than 100 failures, once I merge the pending PRs there should be only 10 left